src/jit/codegenlegacy.cpp

```diff
@@ -19036,16 +19036,31 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                     {
                         noway_assert(callType == CT_USER_FUNC);
 
-                        void* pAddr;
-                        addr = compiler->info.compCompHnd->getAddressOfPInvokeFixup(methHnd, (void**)&pAddr);
-                        if (addr != NULL)
+                        CORINFO_CONST_LOOKUP lookup;
+                        compiler->info.compCompHnd->getAddressOfPInvokeTarget(methHnd, &lookup);
+
+                        void* addr = lookup.addr;
+
+                        _ASSERTE(addr != NULL);
+
+                        bool direct;
+#if defined(_TARGET_ARM_)
+                        direct = compiler->codeGen->validImmForBL((ssize_t)addr);
```

Comment:
`validImmForBL` only make sense for `lookup.accessType == IAT_VALUE`. It should be only called for `lookup.accessType == IAT_VALUE`.